x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/action/TransportPutSnapshotLifecycleAction.java

```diff
@@ -119,49 +120,46 @@ public static class UpdateSnapshotPolicyTask extends AckedClusterStateUpdateTask
         UpdateSnapshotPolicyTask(PutSnapshotLifecycleAction.Request request) {
             super(request, null);
             this.request = request;
-            this.filteredHeaders = Collections.emptyMap();
+            this.filteredHeaders = Map.of();
         }
 
         @Override
         public ClusterState execute(ClusterState currentState) {
-            SnapshotLifecycleMetadata snapMeta = currentState.metadata().custom(SnapshotLifecycleMetadata.TYPE);
+            SnapshotLifecycleMetadata snapMeta = currentState.metadata()
+                .custom(SnapshotLifecycleMetadata.TYPE, SnapshotLifecycleMetadata.EMPTY);
             var currentMode = LifecycleOperationMetadata.currentSLMMode(currentState);
+            final SnapshotLifecyclePolicyMetadata existingPolicyMetadata = snapMeta.getSnapshotConfigurations()
```

Comment:
We now call getSnapshotConfigurations() twice, which creates a new copy of the map. It probably doesn't matter much in the scheme of things.

Reply:
Yeah, I noticed that, too -- there's some stuff in the ILM and SLM sides of this makes me a bit suspicious, so expect a cleanup PR on that front.